reentrancy_guard_example/deposit_example/src/deposit_actor.rs

```diff
@@ -0,0 +1,264 @@
+use actor_library::{
+    context::Context,
+    security::reentrancy_guard::ReentrancyGuard,
+    utils::{map, resolve_address},
+};
+use fil_actors_runtime::{
+    actor_dispatch, actor_error,
+    fvm_ipld_hamt::BytesKey,
+    runtime::{ActorCode, Runtime},
+    ActorDowncast, ActorError, INIT_ACTOR_ADDR,
+};
+use fvm_shared::{address::Address, econ::TokenAmount, error::ExitCode, METHOD_CONSTRUCTOR};
+use num_derive::FromPrimitive;
+use num_traits::Zero;
+
+use crate::state::State;
+
+fil_actors_runtime::wasm_trampoline!(Actor);
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    Balance = frc42_dispatch::method_hash!("Balance"),
+    Deposit = frc42_dispatch::method_hash!("Deposit"),
+    CheckBalance = frc42_dispatch::method_hash!("CheckBalance"),
+    Withdraw = frc42_dispatch::method_hash!("Withdraw"),
+    WithdrawGuarded = frc42_dispatch::method_hash!("WithdrawGuarded"),
+}
+
+pub trait DepositActor {
+    fn constructor(rt: &impl Runtime) -> Result<(), ActorError>;
+
+    fn balance(rt: &impl Runtime) -> Result<TokenAmount, ActorError>;
+
+    fn deposit(rt: &impl Runtime) -> Result<(), ActorError>;
+
+    fn check_balance(rt: &impl Runtime, address: Address) -> Result<TokenAmount, ActorError>;
+
+    fn withdraw(rt: &impl Runtime) -> Result<(), ActorError>;
+
+    fn withdraw_guarded(rt: &impl Runtime) -> Result<(), ActorError>;
+}
+
+pub struct Actor;
+
+impl Context for Actor {}
+
+impl DepositActor for Actor {
+    fn constructor(rt: &impl Runtime) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_is(std::iter::once(&INIT_ACTOR_ADDR))?;
+
+        let st = State::new(rt.store()).map_err(|e| {
+            e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "Failed to create actor state.")
+        })?;
+
+        rt.create(&st)?;
+
+        Ok(())
+    }
+
+    fn balance(rt: &impl Runtime) -> Result<TokenAmount, ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        Ok(rt.current_balance())
+    }
+
+    fn deposit(rt: &impl Runtime) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let value = rt.message().value_received();
```

Comment:
Check this commit https://github.com/Ethernal-Tech/filecoin-actor-library/pull/14/files#diff-7b2f3764cd44c9590de6fd6f3924ce7e97705a53ecc572bb0b5c1e60db8ee118.

```suggestion
        let value = Actor::msg_value();
```

Reply:
Chaged [here](https://github.com/Ethernal-Tech/filecoin-actor-library/blob/ReentrancyGuard_actor_and_tests/reentrancy_guard_example/deposit_example/src/deposit_actor.rs#L71).